src/Symfony/Component/Security/Core/Authentication/Token/NullToken.php

```diff
@@ -33,7 +33,7 @@ public function getCredentials()
 
     public function getUser()
     {
-        return null;
+        return '';
```

Comment:
Should we throw an exception here like we do in `setUser`?

Reply:
Most callers have `if (!token->getUser() instanceof UserInterface)` checks, I think it's good as is

Reply:
I'm kind of mixed on this, if you have multiple user "types" e.g. `User` & `Token` - it's easier to catch an exception. Which in turn falls in line with how the new security system handles `$userBadge->getUser()` || `$passport->getUser()`.